### PR TITLE
remove fields that agent does not accept

### PR DESCRIPTION
**Pull request by paynejacob**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -19,25 +19,26 @@ const (
 )
 
 type K3sConfig struct {
-	ClusterInit bool     `yaml:"cluster-init"`
-	Token       string   `yaml:"token"`
-	Server      string   `yaml:"server"`
-	TLSSan      []string `yaml:"tls-san"`
+	ClusterInit *bool     `yaml:"cluster-init,omitempty"`
```
Comment (dramich) [inline main.go:22]: Why does this need to be a pointer type?

Comment (paynejacob) [reply main.go:22]: so that is can be omitted I am not sure if false is omitted by the yaml serializer.

File: main.go [hunk 2]
```diff
@@ -19,25 +19,26 @@ const (
...
+	Token       string    `yaml:"token,omitempty"`
+	Server      string    `yaml:"server,omitempty"`
+	TLSSan      *[]string `yaml:"tls-san,omitempty"`
```
Comment (dramich) [inline main.go:25]: I like the `omitempty` change

File: main.go (hunk 1 above)
Comment (paynejacob) [reply main.go:22]: it looks like it does not need to be a pointer https://pkg.go.dev/gopkg.in/yaml.v2#section-readme:~:text=Only%20include%20the%20field%20if%20it%27s%20not%20set%20to%20the%20zero%0A%20%20%20%20%20%20%20%20%20%20%20%20%20value

ill update

**Review by dramich — APPROVED** [review]

**Pull request merged**